xrpl-std/src/host/host_bindings.rs

```diff
@@ -0,0 +1,151 @@
+#[link(wasm_import_module = "host_lib")]
+unsafe extern "C" {
```

Comment:
I propose a more folder-oriented structure for `xrpl-std` as seen in this PR. Happy to discuss verbally if anyone dislikes this. There will likely be a lot more in these folders (esp. `core`).